### DISCUSSION TITLE
Extend the API to extract selected_fields  of a connection with the v1/connection path

### DISCUSSION THREAD

**Discussion by jonseymour** [airbytehq/airbyte#45455] [Platform Features or Ideas]
A relatively recent change introduced the ability to set selected fields via the API (ref: #35798 )

However, the API currently lacks the ability to extract selected fields via the matching GET API

This is suboptimal because it means that round-tripping of connection definitions through API is irretrievably lossy - you can CONFIGURE selected fields with the API but you cannot DISCOVER selected fields with the API.

As a result, anyone wishing to maintain connection definitions for use in terraform (for example) is forced to reference schema data from external sources in order to create an artefact that can be used to set the selected fields via the api &/or terraform. 

As it stands, it is not possible, for example, to configure selected fields in the airbyte UI and then export the configured fields via the API because  as it stands, the API doesn't support this use case. Again, it is presumably possible to dig  into airbyte's relational schema, but any need to do so indicates a glaring deficiency in the API itself

**Comment by keyihuang** [comment]
Hi, we applied a fix for this. Now the response of `GET connection` api should include the selected_fields.

**Reply by etiennecallies** [reply]
Hi @keyihuang ! Can you give details about the "fix" you mentioned. Is it on airbyte or terraform provider side? On which version? Thanks

**Reply by salmanahmedjamil** [reply]
Hi @keyihuang please share the details

**Reply by antoineroncoroni-electra** [reply]
Hey, does anyone know which version includes the fix ? I'm on airbyte OSS 1.3.0 and still get empty `selectedFields` when calling the API 😕

**Reply by kev-datams** [reply]
hello @antoineroncoroni-electra, we use Airbyte v1.3.0 with Helm v1.3.1 and all good 👍 
for your Airbyte TF provider, we use `"~> 0.6"`

**Reply by antoineroncoroni-electra** [reply]
Thanks a lot @kev-datams! Good to hear it works on your side, I'll keep investigating on our side

**Comment by kev-datams** [comment]
Hi @jonseymour, did you manage to properly select fields via Terraform as expected ? 🙏

**Reply by etiennecallies** [reply]
Sorry ... I tagged the wrong person. Hi @kev-datams, what would be the airbyte version and airbyte terraform provider version that would theoretically fix this issue please?

**Reply by kev-datams** [reply]
Hi @etiennecallies, I do not know sorry, as I did not try to manage fields selection yet.

**Reply by etiennecallies** [reply]
Thks, no worry
